inception-search-core/src/main/java/de/tudarmstadt/ukp/inception/search/scheduling/IndexScheduler.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.search.scheduling;
+
+import java.util.Iterator;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.inception.search.scheduling.tasks.ReindexTask;
+import de.tudarmstadt.ukp.inception.search.scheduling.tasks.Task;
+
+/**
+ * Indexer scheduler. Does the project reindexing in an asynchronous way.
+ */
+@Component
+public class IndexScheduler
+{
+    private Logger log = LoggerFactory.getLogger(getClass());
+
+    private @Autowired ApplicationContext applicationContext;
+
+    private Thread consumer;
+    private BlockingQueue<Task> queue = new ArrayBlockingQueue<Task>(100);
+
+    @PostConstruct
+    private void startSchedulerThread()
+    {
+        consumer = new Thread(new TaskConsumer(applicationContext, queue), "Index task consumer");
```

Comment:
Is there a reason why you not use Java's `ExecutorService`? It looks like it also allows task scheduling, in addition to that, it has a thread pool and you can get access to the queue behind it.

Reply:
Cf. the comment on the recommender scheduler.